src/client-agent/notify.c

```diff
@@ -39,42 +39,34 @@ char *getsharedfiles()
     return (ret);
 }
 
-#ifndef WIN32
+#define IP_ERROR_TAG "Err"
+
 char *get_agent_ip()
 {
     char agent_ip[IPSIZE + 1] = { '\0' };
-#if defined (__linux__) || defined (__MACH__) || defined (sun) || defined(FreeBSD) || defined(OpenBSD)
-    int sock;
-    int i;
-    static const char * REQUEST = "host_ip";
-
-    for (i = SOCK_ATTEMPTS; i > 0; --i) {
-        if (sock = control_check_connection(), sock >= 0) {
-            if (OS_SendUnix(sock, REQUEST, strlen(REQUEST)) < 0) {
-                mdebug1("Error sending msg to control socket (%d) %s", errno, strerror(errno));
-            }
-            else{
-                if (OS_RecvUnix(sock, IPSIZE, agent_ip) <= 0) {
-                    mdebug1("Error receiving msg from control socket (%d) %s", errno, strerror(errno));
-                    agent_ip[0] = '\0';
-                }
-            }
-
-            close(sock);
-            break;
-        } else {
-            mdebug2("Control module not yet available. Remaining attempts: %d", i - 1);
-            sleep(1);
+#if defined (__linux__) || defined (__MACH__) || defined (sun) || defined(FreeBSD) || defined(OpenBSD) || defined(WIN32)
+    struct sockaddr_storage sas;
+    socklen_t len = sizeof(sas);
+    int err = getsockname(agt->sock, (struct sockaddr *)&sas, &len);
+    
+    if (!err)
+    {
+        switch (sas.ss_family) {
```

Comment:
Adding the default case to this switch and managing it as a case of error would also be prudent. I know the connection should be IPv4 or IPv6, but leaving it without a default case could be a future error case, hard to discover.